autogluon/utils/tabular/ml/models/knn/knn_utils.py

```diff
@@ -0,0 +1,184 @@
+import numpy as np
+from scipy.stats import mode
+from sklearn.utils.extmath import weighted_mode
+import faiss
```

Comment:
Lets move the faiss import into the init calls of the KNeighbors classes so it functions as a lazy import. We want to avoid expanding our required dependencies.

Reply:
Additionally, when this is done, also add a `try_import_faiss` function (See `CatboostModel` for an example of this with `try_import_catboost`. This will help notify users that they need a particular package installed to use the model.

Reply:
Done - thanks!